core/src/main/java/org/apache/accumulo/core/conf/Property.java

```diff
@@ -166,21 +166,6 @@
       PropertyType.STRING,
       "One-line configuration property controlling the network locations "
           + "(hostnames) that are allowed to impersonate other users"),
-  // Crypto-related properties
-  @Experimental
-  INSTANCE_CRYPTO_PREFIX("instance.crypto.opts.", null, PropertyType.PREFIX,
-      "Properties related to on-disk file encryption."),
-  @Experimental
-  @Sensitive
-  INSTANCE_CRYPTO_SENSITIVE_PREFIX("instance.crypto.opts.sensitive.", null, PropertyType.PREFIX,
-      "Sensitive properties related to on-disk file encryption."),
-  @Experimental
-  INSTANCE_CRYPTO_SERVICE("instance.crypto.service",
-      "org.apache.accumulo.core.spi.crypto.NoCryptoService", PropertyType.CLASSNAME,
-      "The class which executes on-disk file encryption. The default does nothing. To enable "
-          + "encryption, replace this classname with an implementation of the"
-          + "org.apache.accumulo.core.spi.crypto.CryptoService interface."),
-
```

Comment:
The 2.0 crypto service concept was based on the idea that crypto policies should be centrally enforced.

Rather than replace the central crypto service with a multitude of separate services per table and WAL, I think you could extend the existing central service concept to support additional config for separate tables and WAL, and to change the implmentation to pass per-table and WAL context information into the central crypto service.

Have you considered this already?

Have you considered extending the 2.0 single-factory concept by supporting the ability to have a single crypto factory/module that can be dropped in and has manages the separate crypto settings based on separate per-table or WAL contexts, rather than requiring separately configured modules? I ask because I think there's value in ensuring that crypto policies are centrally enforced.

Reply:
I did at first but it was easier for me to implement the division of table and WAL initially with them split up. I had problems at first since it was ambiguous in the Accumulo code what the `CryptoService` was doing. The tablet server for example initialized one at startup that could be used for either. I had separate interfaces as well but this led to a lot of duplicate code. Now that I split up the encrypt and decrypt across our mess of internal file builders, it may be easier to go back and combine them into a central service.

Reply:
One issue I had was trying to specify an inner class in the configuration. I am not sure why but this Java notation didn't seem to work:

`org.apache.accumulo.core.spi.crypto.CryptoService$AESGCMCryptoModule`

Any ideas if this can work with our classloading and/or configuration?

Reply:
I think the issue I had was with the way I was loading classes in a previous change. It seems like it should work now with these changes. The only issue is that it is a bit long to configure:
`tserver.walog.crypto.decrypt.service=org.apache.accumulo.core.spi.crypto.AESCryptoService$AESWALCryptoService`